ngraph/core/src/op/range.cpp

```diff
@@ -76,12 +76,18 @@ void op::v4::Range::validate_and_infer_types()
     set_input_is_relevant_to_shape(1);
     set_input_is_relevant_to_shape(2);
 
-    NODE_VALIDATION_CHECK(
-        this, get_input_partial_shape(0).compatible(Shape{}), "'start' input is not a scalar");
-    NODE_VALIDATION_CHECK(
-        this, get_input_partial_shape(1).compatible(Shape{}), "'stop' input is not a scalar");
-    NODE_VALIDATION_CHECK(
-        this, get_input_partial_shape(2).compatible(Shape{}), "'step' input is not a scalar");
+    NODE_VALIDATION_CHECK(this,
```

Comment:
These changes contradicts with the spec. So they are not correct.

Also, take a look at https://github.com/openvinotoolkit/openvino/pull/6008 and https://github.com/openvinotoolkit/openvino/pull/6017 which may have fixed the root cause why the Range got a tensor of shape [1] as input.